### PR TITLE
Feat rc complex layout engine

### PR DESCRIPTION
**Pull request by RyanCCollins**
Add a complex layout engine for setting content layout.

Note this is fairly experimental, but seems to be working really well in my opinion.  @karatechops please do let me know if anything is not working as expected.  I expect to have to iterate a bit on it.  One thing that I would like to do is identify the default properties that should be set when creating a section or a content block and possibly even make that conditional on the content block, that way we are not hardcoding any layout properties into the content blocks themselves.  I would also like to determine some layout presets we can use, but of course that is not a priority right now.

### CODE REVIEW

**Comment by RyanCCollins** [comment]
ETA: I also fixed the vexing `$FlowFixMe` issue and merged that into this branch, as well as added the babel react propType plugin so we can auto generate react propTypes from our flow types.

File: src/js/containers/Dashboard/DashboardPostPage/index.js [hunk 1]
```diff
@@ -29,18 +30,27 @@ import {
...
-  postSectionClearToastMessage
+  postSectionClearToastMessage,
+  postBoxLayoutFormReset,
+  postBoxLayoutFormInput,
+  toggleBoxLayoutForm,
+  postToggleAdvancedLayout
 } from './actions';
-import parseSubmission from './utils';
 import { debounce } from 'grommet-cms/utils';
+import {
```
Comment (RyanCCollins) [inline src/js/containers/Dashboard/DashboardPostPage/index.js]: This container is getting giant.  If we need to add anything else I think it we should make another container

Comment (karatechops) [inline src/js/containers/Dashboard/DashboardPostPage/index.js]: Yep, looks like we're at about max capacity here.

**Comment by karatechops** [comment]
Great work on this PR. The advanced layout engine seems to be working very well. I have not tested all combinations as that would probably take me all day. 😃  I'm sure we'll cover most combinations during our next project with the CMS. 👍

**Pull request merged**